src/Widgets/ControlPanel.hs

```diff
@@ -35,17 +38,33 @@ controlPanel :: Reflex t
              -> Dynamic t Camera
              -> QuaWidget t x (Dynamic t (ComponentState "ControlPanel"))
 controlPanel renderingApi scenarioB selectedObjIdD cameraD = mdo
+    settingsD <- quaSettings
+    eitherReviewSettingsE <- httpGetNowOrOnUpdate $ QuaTypes.reviewSettingsUrl <$> settingsD
+    reviewSettingsD <- Dom.holdDyn (Left "not loaded yet") eitherReviewSettingsE
+    let permsD   = QuaTypes.permissions <$> settingsD
+        showUcD  = QuaTypes.canEditProperties    <$> permsD
+        showAdD  = QuaTypes.canAddDeleteGeometry <$> permsD
+        showGeoD = (&&) <$> showUcD <*> showAdD
+        handleRs (Left _)   = False
+        handleRs (Right rs) = isJust (QtR.reviewsUrl rs) || (not $ null $ QtR.reviews rs)
+        showRevD = handleRs <$> reviewSettingsD
     stateD <- Dom.elDynClass "div" (toClass <$> stateD) $ mdo
-
       -- tab pane
-      (_selTabD, ())
-        <- Dom.elAttr "div" ("style" =: "overflow-y: auto; overflow-x: hidden; height: 100%;") $ do
+      void $ Dom.elAttr "div" ("style" =: "overflow-y: auto; overflow-x: hidden; height: 100%;") $ do
           Dom.elAttr "div" ("style" =: "margin: 0; padding: 0; height: 56px;") Dom.blank
-          runTabWidget $ do
-            addTab "Geometry" $ panelGeometry renderingApi scenarioB selectedObjIdD cameraD
-            addTab "Info" $ panelInfo scenarioB selectedObjIdD
-            addTab "Reviews" panelReviews
-            -- addTab "Services" panelServices
+          let renderTabs showGeo showRev = runTabWidget $ sequence $
+                    geoTab ++ (addTab "Info" $ panelInfo scenarioB selectedObjIdD)
+                    : revTab
+                where
+                  geoTab = if showGeo
```

Comment:
A small nicie: could be done with `when isXXX $ pure tabXXX` to also be easy-changeable between [] and Maybe.